File: Python/Linked List/19. Remove Nth Node From End of List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        dummy = ListNode()
        cur = dummy
        cur.next = head

        first = cur
        second = head

        while n and second:
            second = second.next
            n -= 1

        while second:
            first = first.next
            second = second.next
        
        first.next = first.next.next

        return cur.next

File: Java/Linked List/19. Remove Nth Node From End of List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        
        ListNode dummy = new ListNode();
        ListNode cur = dummy;
        cur.next = head;

        ListNode first = cur;
        ListNode sec = cur;

        while(n>0){
            sec = sec.next;
            n--;
        }

        while(sec.next != null){
            sec = sec.next;
            first = first.next;
        }

        first.next = first.next.next;

        return dummy.next;       

    }
}